### PR TITLE
test: add permission-hook and PTY permission_response tests (#424, #425)

### LINKED ISSUE #425: Add PTY-mode permission_response test for ws-server.js
## Context

From Copilot review on PR #423 ([comment](https://github.com/blamechris/chroxy/pull/423#discussion_r2791125620)).

## Problem

`ws-server.js` now routes `permission_response` in PTY mode (`_handlePtyMessage`), but the existing ws-server tests only cover permission_response resolution in CLI mode. There is no test that validates the round-trip in terminal mode.

## Suggested approach

Add a test that:
1. Starts WsServer in `terminal` mode (with ptyManager/outputParser provided)
2. Performs POST `/permission` to simulate the hook script calling back
3. Sends `permission_response` over WS
4. Asserts the HTTP request resolves with the correct decision

This can be added alongside the existing permission tests in `ws-server.test.js`.

## Related

- #424 (permission-hook.js unit tests)

### PR DESCRIPTION
**Pull request by blamechris**
## Summary

- Add unit tests for the `permission-hook.js` shared module (partial coverage for #424 — withSettingsLock + manager lifecycle)
- Add PTY-mode `permission_response` integration test for ws-server (#425)

Note: register/unregister filesystem tests were removed from this PR to avoid contaminating `~/.claude/settings.json`. Full #424 coverage depends on #429 (configurable settings path).

## Changes

| File | What changed |
|------|-------------|
| `tests/permission-hook.test.js` | New — tests for `withSettingsLock` serialization, error propagation, `createPermissionHookManager` lifecycle |
| `tests/ws-server.test.js` | Added `PTY mode permission_response` describe block — allow/deny resolution, missing requestId handling |

## Test Plan

- [x] All 473 tests pass (16 new, 457 existing)
- [x] No changes to production code

Closes #425
Relates to #424 (partial — remaining coverage blocked on #429)

### CODE REVIEW

File: packages/server/tests/permission-hook.test.js [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+
+    const manager = createPermissionHookManager(emitter)
+    manager.destroy()
+
+    // After destroy, register should still work (lock-wise) but scheduleRetry won't fire
+    // This tests the destroying flag prevents new retries
+    await manager.register()
+    manager.destroy()
+  })
+})
```
[the remark this answers is not in the document]
Comment (blamechris) [reply packages/server/tests/permission-hook.test.js]: **FOLLOW-UP ISSUE**

Created https://github.com/blamechris/chroxy/issues/430 to track this.

**Reason for deferral:** The edge case tests (corrupt JSON, ENOENT directory creation, missing file unregister, retry error emission) all require the module to accept a configurable settings path so tests can write to temp directories. That refactoring is tracked in #429. Once that lands, #430 will add the missing edge case coverage.

**Comment by blamechris** [comment]
## Second-pass review: PR #426

**Verdict: Approve** -- the fix commits addressed all Copilot feedback, tests pass, code is clean.

### What I verified

1. **All 8 Copilot comments resolved.** 4 fixed in commits (`0f336a2`, `c0f7044`, `c773042`, `53d8193`), with `fdee6035` restoring the `join` import that was needed. 4 deferred to follow-up issues #429 and #430.

2. **Unused imports fully cleaned up.** The current import list is:
   - `describe`, `it`, `beforeEach` from `node:test` -- all used
   - `writeFileSync`, `readFileSync` from `fs` -- used for save/restore
   - `join` from `path` -- used on lines 85, 109, 130
   - `homedir` from `os` -- used on lines 85, 109, 130
   - `EventEmitter` from `node:events` -- used in `beforeEach`

3. **Tests pass.** All 13 permission-hook tests and all 3 PTY permission_response tests pass (verified locally with Node 22).

4. **Save/restore pattern analysis.** The settings.json tests use this pattern:
   ```
   settingsBefore = readFileSync(settingsPath)
   // ... register/unregister ...
   writeFileSync(settingsPath, settingsBefore)
   manager.destroy()
   ```
   This works correctly when tests pass. If an assertion throws between register and restore, settings.json is left modified (the chroxy hook remains). This is a known limitation tracked in #429. The risk is low since the hook entry is tagged with `_chroxy: true` and would be cleaned up on next server start/stop. Not blocking.

5. **No cross-test state leakage.** The module-level `_settingsLock` is a promise chain that resolves between tests. Each settings.json test creates its own `createPermissionHookManager` and does its own save/restore. The `beforeEach` resets the EventEmitter. No shared mutable state leaks between test cases.

6. **PTY permission_response tests are well-structured.** They exercise the real HTTP POST `/permission` -> WebSocket broadcast -> client response -> HTTP resolve flow. Both allow and deny paths covered. The missing-requestId test verifies defensive handling of malformed input.

### No new follow-up issues needed

All deferred items are already tracked:
- **#429** -- Configurable settings path for test isolation
- **#430** -- Edge case tests (corrupt JSON, ENOENT, missing file unregister)

No new issues identified in this review pass.

File: packages/server/tests/permission-hook.test.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  // NOTE: register()/unregister()/idempotency tests removed — they wrote to
+  // the real ~/.claude/settings.json and contaminated other running sessions.
+  // These will return once #429 lands a configurable settingsPath parameter.
+
+  it('destroy() cancels pending retry timers', async () => {
+    const manager = createPermissionHookManager(emitter)
+    // destroy() should be safe to call even without register()
+    manager.destroy()
+    // No assertion needed — just verify it doesn't throw
+  })
```
[the remark this answers is not in the document]
Comment (blamechris) [reply packages/server/tests/permission-hook.test.js]: **FIX**

Fixed in `d74f8a82`

**Change:** Renamed test to accurately describe what it verifies — that destroy() is safe to call without a prior register().

```diff
-  it('destroy() cancels pending retry timers', async () => {
-    const manager = createPermissionHookManager(emitter)
-    // destroy() should be safe to call even without register()
-    manager.destroy()
-    // No assertion needed — just verify it doesn't throw
+  it('destroy() does not throw when called without register()', async () => {
+    const manager = createPermissionHookManager(emitter)
+    manager.destroy()
+    // Verifies destroy() is safe to call before any registration
```
